crates/ra_syntax/src/ptr.rs

```diff
@@ -64,6 +68,12 @@ impl<N: AstNode> PartialEq for AstPtr<N> {
     }
 }
 
+impl<N: AstNode> Hash for AstPtr<N> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.raw.hash(state)
```

Comment:
what's the difference to the derived implementation here? :thinking:

Reply:
Derive would be `impl<N: AstNode + Hash>`